tls/resource_certificate.go

```diff
@@ -185,6 +199,26 @@ func createCertificate(d *schema.ResourceData, template, parent *x509.Certificat
 	}
 	certPem := string(pem.EncodeToMemory(&pem.Block{Type: pemCertType, Bytes: certBytes}))
 
+	private_key, err := parsePrivateKey(d, "private_key_pem", "key_algorithm")
+	// Set PKCS12 data. This is only set if there is a private key present.
+	if err != nil {
+		cert, err := x509.ParseCertificate(certBytes)
+		if err != nil {
+			return err
+		}
+		caCerts = append(caCerts, parent)
```

Comment:
I tried this out and this gave me troubles in the `PFX` validation.
If the parent (which in this case is the same as the cert as `cert`) is provided the p12/pkcs12 is available but testing it with `openssl pkcs12 -info` gave me errors:
```
CRYPTO_internal:header too long
CRYPTO_internal:bad object header
CRYPTO_internal:nested asn1 error
CRYPTO_internal:decode error
```

Leaving `caCerts` empty, gave me a valid p12/pkcs12.

Reply:
removed